pkg/alertmanager/amcfg.go

```diff
@@ -128,6 +128,58 @@ func validateConfigInputs(am *monitoringv1.Alertmanager) error {
 	return nil
 }
 
+func (cg *configGenerator) generateGlobalConfig(
+	ctx context.Context,
+	baseConfig *alertmanagerConfig,
+	am *monitoringv1.Alertmanager,
+	amConfigs map[string]*monitoringv1alpha1.AlertmanagerConfig,
+) error {
```

Comment:
The global `AlertmanagerConfig` object might not exist in the amConfigs map (for instance, `spec.alertmanagerConfigNamespaceSelector` might not select the Alertmanager's namespace). It should be up to the caller to retrieve the `AlertmanagerConfig` object, handle the error case and pass it to this method.

```suggestion
func (cg *configGenerator) generateGlobalConfig(
	ctx context.Context,
	amConfig *monitoringv1alpha1.AlertmanagerConfig,
) (*alertmanagerConfig, error) {
```